includes/class-wc-stripe-webhook-handler.php

```diff
@@ -1136,12 +1136,29 @@ public function process_setup_intent( $notification ) {
 			return;
 		}
 
-		if ( ! $order->has_status(
-			apply_filters(
-				'wc_gateway_stripe_allowed_payment_processing_statuses',
-				[ OrderStatus::PENDING, OrderStatus::FAILED ]
-			)
-		) ) {
+		$allowed_payment_processing_statuses = apply_filters_deprecated(
+			'wc_gateway_stripe_allowed_payment_processing_statuses',
+			[ [ OrderStatus::PENDING, OrderStatus::FAILED ] ],
+			'9.6.0',
+			'wc_stripe_allowed_payment_processing_statuses',
+			'The wc_gateway_stripe_allowed_payment_processing_statuses filter is deprecated since WooCommerce Stripe Gateway 9.6.0, and will be removed in a future version.'
+		);
+
+		/**
+		 * Filters the valid order statuses for payment processing.
+		 *
+		 * @since 9.6.0
+		 *
+		 * @param array $allowed_payment_processing_statuses The allowed payment processing statuses from the deprecated filter.
```

Comment:
As above, I think we shouldn't document the deprecated filter.
```suggestion
		 * @param array $allowed_payment_processing_statuses The allowed payment processing statuses.
```